deepinv/datasets/lidc_idri.py

```diff
@@ -0,0 +1,139 @@
+from typing import (
+    Any,
+    Callable,
+    NamedTuple,
+    Optional,
+)
+import os
+from pydicom import dcmread
+import torch
+import numpy as np
+import pandas as pd
+
+
+class LidcIdriSliceDataset(torch.utils.data.Dataset):
+    """Dataset for `LIDC-IDRI <https://www.cancerimagingarchive.net/collection/lidc-idri/>`_ that provides access to CT image slices.
+
+    | The Lung Image Database Consortium image collection (LIDC-IDRI) consists
+    | of diagnostic and lung cancer screening thoracic computed tomography (CT)
+    | scans with marked-up annotated lesions.
+    | It is a web-accessible international resource for development, training,
+    | and evaluation of computer-assisted diagnostic (CAD) methods for lung
+    | cancer detection and diagnosis.
+    | To download the dataset, you will need to install the `NBIA Data Retriever <https://wiki.cancerimagingarchive.net/display/NBIA/Downloading+TCIA+Images>`_.
+
+    **Raw data file structure:** ::
+
+        self.root --- LIDC-IDRI --- LICENCE
+                   |             -- LIDC-IDRI-0001 --- `STUDY_UID` --- `SERIES_UID` --- xxx.xml
+                   |             |                                                   -- 1-001.dcm
+                   |             -- LIDC-IDRI-1010                                   |
+                   |                                                                 -- 1-xxx.dcm
+                   -- metadata.csv
+
+    :param str root: Root directory of dataset. Directory path from where we load and save the dataset.
+    :param callable, optional transform: TODO
+
```

Comment:
How are the slices loaded? Please give more information here. What are the sizes of the slices?

Reply:
we open a dicom file with pydicom and we retrieve the "pixel_array" field

Reply:
ct scan folders are found thanks to the metadata.csv
and in these folders, we have the dicom files